kong/plugins/key-auth/handler.lua

```diff
@@ -149,6 +150,11 @@ end
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- check if preflight request and wether it should be authenticated
+  if conf.authenticate_preflight and get_method() == "OPTIONS" then
```

Comment:
It's actually a bug, hadn't checked CI results yet. It should have been

`if not conf.authenticate_preflight and ...`

so we are at scenario 2. I prefer this wording over the CORS wording of `preflight_continue`, so maybe update CORS as well?